cmd/zed/permission.go

```diff
@@ -106,27 +107,29 @@ func checkCmdFunc(cmd *cobra.Command, args []string) error {
 		return err
 	}
 
-	request := &v0.CheckRequest{
-		TestUserset: &v0.ObjectAndRelation{
-			Namespace: stringz.Join("/", token.System, objectNS),
-			ObjectId:  objectID,
-			Relation:  relation,
+	request := &v1.CheckPermissionRequest{
+		Resource: &v1.ObjectReference{
+			ObjectType: nsPrefix(objectNS, token.System),
+			ObjectId:   objectID,
 		},
-		User: &v0.User{UserOneof: &v0.User_Userset{
-			Userset: &v0.ObjectAndRelation{
-				Namespace: stringz.Join("/", token.System, subjectNS),
-				ObjectId:  subjectID,
-				Relation:  subjectRel,
+		Permission: relation,
+		Subject: &v1.SubjectReference{
+			Object: &v1.ObjectReference{
+				ObjectType: nsPrefix(subjectNS, token.System),
+				ObjectId:   subjectID,
 			},
-		}},
+			OptionalRelation: subjectRel,
+		},
 	}
 
-	if zedToken := cobrautil.MustGetString(cmd, "revision"); zedToken != "" {
-		request.AtRevision = &v0.Zookie{Token: zedToken}
+	if zedtoken := cobrautil.MustGetString(cmd, "revision"); zedtoken != "" {
```

Comment:
Maybe we extract this into a util method and use everywhere?

Reply:
I think I'll just create an issue for "adding a consistency flag" and we can do that in a follow up.